recipes/ffmpeg/all/conanfile.py

```diff
@@ -521,71 +527,97 @@ def opt_append_disable_if_set(args, what, v):
+
+        if is_apple_os(self):
+            apple_min_version_flag = AutotoolsToolchain(self).apple_min_version_flag
+            if apple_min_version_flag:
+                extra_cflags.append(apple_min_version_flag)
+                extra_ldflags.append(apple_min_version_flag)
+        if is_msvc(self):
+            pkg_config = os.getenv("PKG_CONFIG") # TODO
+            args.append(f"--pkg-config={pkg_config}")
             args.append("--toolchain=msvc")
-            if self.settings.compiler == "Visual Studio" and tools.Version(self.settings.compiler.version) <= "12":
+            if self.settings.compiler == "Visual Studio" and Version(self.settings.compiler.version) <= "12":
                 # Visual Studio 2013 (and earlier) doesn't support "inline" keyword for C (only for C++)
-                self._autotools.defines.append("inline=__inline")
+                tc.extra_defines.append("inline=__inline")
         if cross_building(self):
             if self._target_os == "emscripten":
                 args.append("--target-os=none")
             else:
-                args.append("--target-os={}".format(self._target_os))
+                target_os = "darwin" if is_apple_os(self) else str(self.settings.os)
+                args.append(f"--target-os={target_os}")
 
             if is_apple_os(self):
                 if self.options.with_audiotoolbox:
-                    args.append("--disable-outdev=audiotoolbox")
+                    args2["--disable-outdev"] = "audiotoolbox"
 
-                xcrun = tools.XCRun(self.settings)
-                apple_arch = tools.to_apple_arch(str(self.settings.arch))
+                xcrun = XCRun(self)
+                apple_arch = to_apple_arch(self)
                 extra_cflags.extend(
-                    ["-arch {}".format(apple_arch), "-isysroot {}".format(xcrun.sdk_path)])
+                    [f"-arch {apple_arch}", f"-isysroot {xcrun.sdk_path}"])
                 extra_ldflags.extend(
-                    ["-arch {}".format(apple_arch), "-isysroot {}".format(xcrun.sdk_path)])
-
-        args.append("--extra-cflags={}".format(" ".join(extra_cflags)))
-        args.append("--extra-ldflags={}".format(" ".join(extra_ldflags)))
-
-        self._autotools.configure(
-            args=args, configure_dir=self._source_subfolder, build=False, host=False, target=False)
-        return self._autotools
+                    [f"-arch {apple_arch}", f"-isysroot {xcrun.sdk_path}"])
+
+        if len(extra_cflags) > 0:
+            args2["--extra-cflags"]  = " ".join(extra_cflags)
+        if len(extra_ldflags) > 0:
+            args2["--extra-ldflags"] = " ".join(extra_ldflags)
+
+        # these aregs aren't understood by ffmpeg's configure script
+        args2["--sbindir"] = None
+        args2["--includedir"] = None
+        args2["--oldincludedir"] = None
+
+        tc.configure_args.extend(args)
+        tc.update_configure_args(args2)
+
+        tc.generate()
+
+
+        if is_msvc(self):
```

Comment:
I stole this from `recipes/coin-utils/all/conanfile.py` maybe see if you can build that one?


```
        env = tc.environment()
        if is_msvc(self):
            compile_wrapper = unix_path(self, self.conf.get("user.automake:compile-wrapper", check_type=str))
            ar_wrapper = unix_path(self, self.conf.get("user.automake:lib-wrapper", check_type=str))
            env.define("CC", f"{compile_wrapper} cl -nologo")
            env.define("CXX", f"{compile_wrapper} cl -nologo")
            env.define("LD", "link -nologo")
            env.define("AR", f"{ar_wrapper} \"lib -nologo\"")
            env.define("NM", "dumpbin -symbols")
            env.define("OBJDUMP", ":")
            env.define("RANLIB", ":")
            env.define("STRIP", ":")
        tc.generate(env)
```